Sources/Utility/ArrayExtensions.swift

```diff
@@ -20,10 +20,10 @@ extension Array {
         var t = [T]()
         var u = [U]()
         for e in self {
-            if e is T {
-                t.append(e as! T)
-            } else {
-                u.append(e as! U)
+            if let e = e as? T {
+                t.append(e)
+            } else if let e = e as? U {
+                u.append(e)
```

Comment:
Works for me
```swift
extension Array {
    public func partition<T, U>() -> ([T], [U]) {
        var t = [T]()
        var u = [U]()
        for e in self {
            if case let e as T = e {
                t.append(e)
            } else if case let e as U = e {
                u.append(e)
            }
        }
        return (t, u)
    }
}

let a = [1, 2, 3, "2", "3"]
let partition: ([Int], [String]) = a.partition()
```

agreed though that `if case let` looks heavier at first but I have grown to like it lol